src/Angor/Avalonia/Angor.Wallet/Domain/XPubCollection.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+
+namespace Angor.Wallet.Domain;
+
+public sealed record XPubCollection : IEnumerable<XPub>
+{
+    private readonly Dictionary<DomainScriptType, XPub> xpubs;
+
+    private XPubCollection(Dictionary<DomainScriptType, XPub> xpubs)
+    {
+        this.xpubs = xpubs;
+    }
+
+    public static XPubCollection Create(XPub segwitXPub, XPub taprootXPub)
```

Comment:
Is this used for testing? seems very specific

Reply:
No, this is not just for testing - let me explain the design:

XPubCollection contains the extended public keys that define what a wallet can do. We need this specific implementation because the Infrastructure layer needs these xpubs to:
   - Derive addresses
   - Track UTXOs
   - Calculate balances
   - Monitor transactions

To simplify things even more I've changed XPubCollection to derive from Collection<XPub> to remove unnecessary complexity, while still retaining its domain meaning and purpose.